src/coreclr/jit/morph.cpp

```diff
@@ -3264,29 +3222,31 @@ void Compiler::fgMakeOutgoingStructArgCopy(GenTreeCall* call, CallArg* arg)
         assert(!fgGlobalMorph);
     }
 
+    call->gtArgs.SetNeedsTemp(arg);
```

Comment:
This was previously done in `ArgsComplete` (the code in the `GTF_ASG` case), but it makes more sense just to do it immediately when we introduce the temp.